extensions/interactions/NumberWithUnits/directives/NumberWithUnitsRulesServiceSpec.js

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for Number with Units rules.
+ */
+
+describe('Number with Units rules service', function() {
+  beforeEach(module('oppia'));
+
+  var nurs = null;
+  beforeEach(inject(function($injector) {
+    nurs = $injector.get('numberWithUnitsRulesService');
+  }));
+
+  var createNumberWithUnitsDict = function(
+      type, real, fraction, units) {
+    return {
+      type: type,
+      real: real,
+      fraction: fraction,
+      units: units
+    };
+  };
+
+  var REAL_RULE_INPUT = {
+    f: {
```

Comment:
Why not use `createNumberWithUnitsDict` function here?

Reply:
Yes, we can also use it here.